### PR TITLE
Object store improvements and bugfixes

### LINKED ISSUE #545: Object store (S3/TOS/OSS) production ready - Q3
Object store has some missing bug fix and feature such as attach support. They will all be included in 0.2.0 release.

For the bug, it changes the data location in S3, we develop a tools for data migration, the tool needs some doc. https://github.com/ByConity/ByConity/pull/490

### PR DESCRIPTION
**Pull request by gaodayue**
### Changelog category :
- Bug Fix

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixes #545 
- added attach/detach support for s3
- fix incorrect data path and part metadata

### Documentation entry for user-facing changes

### CODE REVIEW

**Comment by gaodayue** [comment]
A notice on configuration chagne: user needs to include `<storage_configuration>` for daemon manager after this PR, refer to the example config in [docker/ci-deploy/config/daemon-manager.yml](https://github.com/ByConity/ByConity/pull/614/files#diff-8a43b386c15a3fc43b30da53b9b6b2e51a6946765da2b6d7d9d8fea5883a69b9)

File: src/Catalog/MetastoreCommon.h [hunk 1]
```diff
@@ -78,9 +78,10 @@ struct BatchCommitRequest
         : with_cas(with_cas_), allow_cas_fail(allow_cas_fail_)
     {
     }
-    
+
     void AddPut(const SinglePutRequest & put) { puts.emplace_back(put); }
```
Comment (hustnn) [inline src/Catalog/MetastoreCommon.h:82]: Addput can also be unified later.

**Review by hustnn — APPROVED** [review]

**Comment by hustnn** [comment]
> A notice on configuration chagne: user needs to include `<storage_configuration>` for daemon manager after this PR, refer to the example config in [docker/ci-deploy/config/daemon-manager.yml](https://github.com/ByConity/ByConity/pull/614/files#diff-8a43b386c15a3fc43b30da53b9b6b2e51a6946765da2b6d7d9d8fea5883a69b9)

cc @dmthuc @weigangchen1 @vivek-debug

**Comment by gaodayue** [comment]
please also be aware that this PR only adds `attach/detach partition` support for s3 CnchMergeTree **without** `UNIQUE KEY`.

**Pull request merged**